File: Sqrt-x-Square-Root.py
class Solution:
    def mySqrt(self, n: int) -> int:
        l=1
        h=n
        while(h>=l):
            mid=(l+h)//2
            if (mid*mid<=n):
                l=mid+1
            else:
                h=mid-1
        return h
        

File: Sqrt-x-Square-Root.js
/**
 * @param {number} n
 * @returns {number}
 */

class Solution {
    floorSqrt(x) {
        // code here
        let l=1
        let h=x
        while (h>=l){
            let mid=Math.floor((l+h)/2)
            if (mid*mid <= x){
                l=mid+1
            }
            else{
                h=mid-1
            }
        }return h
    }
}
